chain/manager.go

```diff
@@ -513,7 +513,8 @@ func (m *Manager) revalidatePool() {
 		delete(m.txpool.indices, txid)
 	}
 	m.txpool.ms = consensus.NewMidState(m.tipState)
-	txns := append(m.txpool.txns, m.txpool.lastReverted...)
+	txns := m.txpool.txns
```

Comment:
```go
var txns []transactions
txns = append(txns, m.txpool.tns)
txns = append(txns, m.txpool.lastReverted)
```

is probably a bit safer. `append` might reuse the memory of the thing you append to so starting with an empty slice is probably better. Same for potential other spots.

Reply:
reusing memory is intentional here; this is a "filter" pattern where we want to discard the invalid transactions present in `m.txpool.txns`. We do that by reslicing to 0 and appending.

Usually, the filter pattern looks like this:
```go
rem := m.txpool.txns[:0]
for _, txn := range m.txpool.txns {
    if valid(txn) {
        rem = append(rem, txn)
    } 
}
m.txpool.txns = rem
```
But in this case, since we want to process `lastReverted` as well, I switched things up a little.

If we must keep this silly linter rule, we can rewrite this as:
```go
m.txpool.txns = append(m.txpool.txns, m.txpool.lastReverted...)
rem := m.txpool.txns[:0]
for _, txn := range m.txpool.txns {
    if valid(txn) {
        rem = append(rem, txn)
    } 
}
m.txpool.txns = rem
```
or we could duplicate the loop, i.e. loop over `m.txpool.txns`, then over `m.txpool.lastReverted`. This would be more efficient (since it saves an append), but more verbose.